nautobot/docs/development/apps/api/platform-features/populating-extensibility-features.md

```diff
@@ -59,3 +63,76 @@ config = AnimalSoundsConfig
     It is crucial that you add the `sender=self` parameter to the `connect` method call - otherwise your signal handler will run as many times as the `NautobotAppConfig.ready` method is called, which may be a lot of times.
 
 After writing this code, run `nautobot-server migrate` or `nautobot-server post_upgrade`, then restart the Nautobot server, and you should see that this custom Relationship has now been automatically created.
+
+## Advanced
+
+While the signal handler approach works for most use cases, it does have its downsides. Other options for the population of base state are available, each with their own trade-offs:
+
+- Signal Handlers on `nautobot_database_ready` (see above)
+- [Django Onboard Tooling](https://docs.djangoproject.com/en/4.2/howto/initial-data/)
+    - [Django Data Migrations](https://docs.djangoproject.com/en/4.2/topics/migrations/#data-migrations)
+    - [Django Fixtures](https://docs.djangoproject.com/en/4.2/topics/db/fixtures/#fixtures)
+- [Design Builder Jobs](https://docs.nautobot.com/projects/design-builder/en/latest/)
+- Creating data in-place where needed
+
+### Signal Handlers On `nautobot_database_ready`
+
+**Pro**:
+
+- Execution: Automatic
+- Implementation: Easy
+
+**Con**:
+
+- Prevent Nautobot from starting when there are errors
+
+### Django Data Migrations
+
+**Pro**:
+
+- Execution: Automatic
+- Provide a clean way to delete any associated objects when uninstalling the app
+
+**Con**:
+
+- Implementation: Medium
+- Prevent Nautobot from starting when there are errors
+- Immutable—if you add new objects, you need to add a new migration
+- Can't easily be feature-toggled (if you include settings lookups in your migration and later change those settings, the migrations will _not_ run again)
+
+### Django Fixtures
+
+**Pro**:
+
+- Have good support for usage in unit tests
+- Have good performance
+
+**Con**:
+
+- Execution: Manual
+- Implementation: Medium
+- Creating/updating them is not straight-forward, especially for big data sets
+
+### Design Builder Jobs
+
+**Pro**:
+
+- Implementation: Easy
+
+**Con**:
+
+- Execution: Manual
```

Comment:
- The possibility of later user modifications to data must be accounted for and handled in code.